tests/src/artifact/tar/tar_test.cpp

```diff
@@ -205,9 +209,28 @@ TEST_F(TarTestEnv, TestTarReaderExtraDataAtEOF) {
 
 	mender::tar::ExpectedEntry next_tar_entry = tar_reader.Next();
 
-	EXPECT_FALSE(next_tar_entry);
+	ASSERT_FALSE(next_tar_entry);
 	EXPECT_EQ(next_tar_entry.error().code, tar::MakeError(tar::TarExtraDataError, "").code);
-	EXPECT_EQ(next_tar_entry.error().message, "Reached the end of the archive: ");
+	EXPECT_EQ(
+		next_tar_entry.error().message,
+		"Reached the end of the archive: Only zero bytes allowed after an end of archive");
+}
+
+TEST_F(TarTestEnv, TestTarReaderExtraZeroesAtEOF) {
+	std::ifstream fs {tmpdir->Path() + "/test-large-with-zeroes.tar"};
+	mender::common::io::StreamReader sr {fs};
+	mender::tar::Reader tar_reader {sr};
+	mender::tar::ExpectedEntry tar_entry = tar_reader.Next();
+	ASSERT_TRUE(tar_entry);
+
+	io::Discard discard;
+	EXPECT_EQ(io::Copy(discard, tar_entry.value()), error::NoError);
+
+	mender::tar::ExpectedEntry next_tar_entry = tar_reader.Next();
```

Comment:
As long as it is only 0s, this second read should return `error::NoError`, like any other read to a tar that has no trailing zeros.

By looking at the code you wrote, that is what I expect too, so I cannot see the bug there.

Reply:
It returns `TarEOFError` because there is no other entry. Like with any other tar that has no trailing zeroes. Or am I missing something?

Reply:
You are right, my apologies. I took the wrong test as a reference - what we want is this test to behave the same as `TestTarReaderEOF` and that is exactly what it is doing :+1: